content/articles/how-to-build-a-quiz-app-with-vanilla-javascript-and-tailwind-css/index.md

```diff
@@ -0,0 +1,409 @@
+    while (answerButtonsElement.firstChild) {
+        answerButtonsElement.removeChild(answerButtonsElement.firstChild)
+    }
+}
+```
+In the function above, we ensured a `next` button appears as soon as the user clicks on an answer and immediately disappears when clicked as a new question surfaces.
+
+We need to create the function to select an answer when one of the option buttons is clicked.
+
+#### The `chooseAnswer` function
+First, you have to target the answer button the user clicks on by creating a variable like this: `const selectedButton = e.target`.Then, you can just get a variable `correct` which will be equal `selectedButton.dataset.correct`.
+
+Next, you need to loop through the other buttons and set the class for them. You are going to create an array from the `answerButtonsElement.children`. The `answerButtonsElement.children` is returning a live collection, and it updates on its own, so you have to convert it to an array to make it easier to loop through each button with the `forEach` function.
+
+You then call the `setStatusClass` function, which we will create in a bit and pass in `button.dataset.correct`, determining if the answer picked is the right one.
+
+You can also use a conditional statement to convert the `next` button to a `restart` button if the user gets to the last question. The test then restarts once the `restart` button is clicked.  Here is the code for the `chooseAnswer` function:
+
+```js
+const chooseAnswer = (e) => {
+    const selectedButton = e.target
+    const correct = selectedButton.dataset.correct
+    Array.from(answerButtonsElement.children).forEach(button => {
+        setStatusClass(button, button.dataset.correct)
+    })
+    if (shuffledQuestions.length > currentQuestionIndex + 1) {
+        nextButton.classList.remove('hide')
+    } else {
+        startButton.innerText = 'Restart'
+        startButton.classList.remove('hide')
+    }
+}
+
+```
+
+#### The `setStatusClass` function
+First thing you need to do is to call the `ClearStatusClass` function which we will create after this and pass in `element`. Then you use a conditional statement to actually check if the answer picked is correct. If it is, add the `correct` class, else, add the `wrong` class.
+
+Below is the code:
+
+```js
+const setStatusClass = (element, correct) => {
+    clearStatusClass(element)
+    if (correct) {
+        element.classList.add('correct')
+    } else {
+        element.classList.add('wrong')
+    }
+}
+
+```
+`clearStatusClass` function code is as shown below:
+
+```js
+const clearStatusClass = (element) => {
+    element.classList.remove('correct')
+    element.classList.remove('wrong')
+}
+```
+
+Below is a video of our quiz application in action:
+
+<iframe width="727" height="409" src="https://www.youtube.com/embed/ENdP7aFVFpY" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```

Comment:
if you want to add a video for your article, you can screen record it then add it to your markdown as a `.gif`. The `<iframe>` doesn't seem to have worked.